File: core/src/nl/t64/game/rpg/screens/menu/MenuCredits.java
package nl.t64.game.rpg.screens.menu;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.utils.Align;
import com.badlogic.gdx.utils.ScreenUtils;


public class MenuCredits extends MenuScreen {

    private static final String CREDITS_PATH = "licenses/credits.txt";
    private static final int SCROLL_SPEED = 100;
    private static final int EMPTY_LINES = 20;

    private final ScrollPane scrollPane;

    public MenuCredits() {
        String emptyLines = System.lineSeparator().repeat(EMPTY_LINES);
        String creditsText = emptyLines + Gdx.files.internal(CREDITS_PATH).readString() + emptyLines;
        var textStyle = new Label.LabelStyle(menuFont, Color.BLACK);
        var credits = new Label(creditsText, textStyle);
        credits.setAlignment(Align.top | Align.center);
        credits.setWrap(true);
        this.scrollPane = new ScrollPane(credits);
        this.createTable();
    }

    @Override
    void setupScreen() {
        scrollPane.setScrollY(0);
        scrollPane.updateVisualScroll();
        applyListeners();
        stage.addActor(table);
    }

    @Override
    public void render(float dt) {
        ScreenUtils.clear(Color.BLACK);
        stage.act(dt);
        scrollPane.setScrollY(scrollPane.getScrollY() + dt * SCROLL_SPEED);
        stage.draw();
    }

    private void createTable() {
        table = new Table();
        table.center();
        table.setFillParent(true);
        table.defaults().width(Gdx.graphics.getWidth());
        table.add(scrollPane);
    }

    private void applyListeners() {
        stage.addListener(new ListenerKeyConfirm(super::processBackButton));
        stage.addListener(new ListenerKeyCancel(super::processBackButton));
    }

}

File: core/src/nl/t64/game/rpg/screens/menu/MenuCredits.kt
package nl.t64.game.rpg.screens.menu

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Label.LabelStyle
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.utils.Align
import com.badlogic.gdx.utils.ScreenUtils


private const val CREDITS_PATH = "licenses/credits.txt"
private const val SCROLL_SPEED = 100
private const val EMPTY_LINES = 20

class MenuCredits : MenuScreen() {

    private val scrollPane: ScrollPane

    init {
        val emptyLines = System.lineSeparator().repeat(EMPTY_LINES)
        val creditsText = emptyLines + Gdx.files.internal(CREDITS_PATH).readString() + emptyLines
        val textStyle = LabelStyle(menuFont, Color.BLACK)
        val credits = Label(creditsText, textStyle)
        credits.setAlignment(Align.top or Align.center)
        credits.wrap = true
        scrollPane = ScrollPane(credits)
        createTable()
    }

    override fun setupScreen() {
        scrollPane.scrollY = 0f
        scrollPane.updateVisualScroll()
        applyListeners()
        stage.addActor(table)
    }

    override fun render(dt: Float) {
        ScreenUtils.clear(Color.BLACK)
        stage.act(dt)
        scrollPane.scrollY = scrollPane.scrollY + dt * SCROLL_SPEED
        stage.draw()
    }

    private fun createTable() {
        table = Table()
        table.center()
        table.setFillParent(true)
        table.defaults().width(Gdx.graphics.width.toFloat())
        table.add(scrollPane)
    }

    private fun applyListeners() {
        stage.addListener(ListenerKeyConfirm { super.processBackButton() })
        stage.addListener(ListenerKeyCancel { super.processBackButton() })
    }

}